src/ansi.rs

```diff
@@ -672,6 +673,70 @@ pub trait Handler {
 
     /// Set mouse cursor icon.
     fn set_mouse_cursor_icon(&mut self, _: CursorIcon) {}
+
+    /// Report current keyboard mode.
+    ///
+    /// ## Caveat
+    ///
+    /// This request must be done successively with the [`identify_terminal`] request, given
+    /// clients rely on the later to identify whether the terminal suppors kitty's keyboard
+    /// protocol.
+    ///
+    /// [`identify_terminal`]: Self::identify_terminal
```

Comment:
This technically isn't actually true. The whole `identify_terminal` thing is just a hack for applications to figure out if an escape sequence is not supported, or if there just hasn't been a reply yet. Basically just a "sync point" and other escapes could work too.

This is very much a general thing, even though this protocol specifically incorporates it, so I don't think this section is necessary. People curious about the exact correct behavior are probably better off reading the protocol itself anyway.